src/kv/KeyValueDB.h

```diff
@@ -161,6 +161,11 @@ class KeyValueDB : public PriorityCache::PriCache {
   // vector cfs contains column families to be created when db is created.
   virtual int create_and_open(std::ostream &out,
 			      const vector<ColumnFamily>& cfs = {}) = 0;
+
+  virtual int open_read_only(ostream &out, const vector<ColumnFamily>& cfs = {}) {
```

Comment:
Can you clarify in the commit message what the specific problem with opening read-write is?  Is it that bluefs might trigger a compaction at a bad time?  Or rocksdb might compact?

Reply:
@liewegas - done